src/sds_data_model/_vector.py

```diff
@@ -716,7 +716,10 @@ def _get_name(
     if name:
         return name
     elif metadata:
-        return metadata.title
+        if metadata.title:
+            return metadata.title
+        else: 
+            ValueError("Metadata provided, but no title present, a name must be supplied.")
     else:
         raise ValueError("If there isn't any metadata, a name must be supplied.")
```

Comment:
`mypy` doesn't seem to recognise the nested returns here and is complaining. One solution would to nested assignments followed by a final return:

```suggestion
    if name:
        _name = name
    elif metadata:
        if metadata.title:
            _name = metadata.title
        else: 
            ValueError("Metadata provided, but no title present, a name must be supplied.")
    else:
        raise ValueError("If there isn't any metadata, a name must be supplied.")
    return _name
```